### DISCUSSION TITLE
Uno.sdk TargetFramework Extension Support and Uno.ui Multi Platform Support Issues

### DISCUSSION THREAD

**Discussion by HaoLife** [unoplatform/uno#17064] [Ideas]
Uno.ui supports multiple platforms. When using wasm and skia, when using the nuget package to restore, replace the DLL file of the netx.x platform to support it

Uno.sdk supports the use of extended TargetFramework for multi platform implementation.

I have a question, why not extend TargetFramework on SDK and use this extension's SDK (framework. sdk as the basic TargetFramework extension) for uno.ui.

[MSBuildApp.zip](https://github.com/user-attachments/files/15750775/MSBuildApp.zip)

**Comment by HaoLife** [comment]
Uno.ui has the following inconveniences when using wasm or skia.

1. The source code f12 and the final debugging DLL package are different, making it inconvenient to view the source code and debug
3. When extending uno.ui, it is inconvenient when it involves support for different platforms such as wasm or skia (such as the z-axis, which prevents access to existing FrameworkElement member extensions)

**Reply by jeromelaban** [reply]
@HaoLife we intend to do this in a future release, but we're not there yet. It's part of a refactoring that implied moving to net9, which is was done recently. It also has implications for backward compatibility (the "reference" API still needs to be present).

**Reply by HaoLife** [reply]
thanks
